### PR TITLE
Attempt to make thumb creation more tolerant to bug

### PR DESCRIPTION
**Pull request by frankrousseau**
- Make thumb module more modular
- Try to build thumb only for jpg and png files
- remove file open step before thumbnail generation (useless with a create write stream)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,9 +19,10 @@
     "cozy-realtime-adapter": "0.11.4",
     "imagemagick": "0.1.3",
     "jade": "1.3.1",
+    "mime": "^1.2.11",
```
Comment (aenario) [inline package.json]: lock versions

**Comment by aenario** [comment]
why use mime & not just check file name 's extensions ?

mime does just that : 
https://github.com/broofa/node-mime/blob/master/mime.js#L74
https://github.com/broofa/node-mime/blob/master/types/mime.types#L1319
https://github.com/broofa/node-mime/blob/master/types/mime.types#L1324

**Comment by aenario** [comment]
lock printit version too

**Comment by frankrousseau** [comment]
`mime` is more expressive.

**Comment by aenario** [comment]
Ok, we can reconsider when moving this into the DS. 
LGTML :+1:

**Pull request merged**